pkg/webhook/webhook.go

```diff
@@ -129,13 +130,81 @@ func deserializePod(ar *v1beta1.AdmissionReview) (corev1.Pod, error) {
 	/* unmarshal Pod from AdmissionReview request */
 	pod := corev1.Pod{}
 	err := json.Unmarshal(ar.Request.Object.Raw, &pod)
-	/* fix for missing "default" namespace */
-	if pod.ObjectMeta.Namespace == "" {
-		pod.ObjectMeta.Namespace = "default"
+	if pod.ObjectMeta.Namespace != "" {
+		return pod, err
+	}
+	ownerRef := pod.ObjectMeta.OwnerReferences
+	if ownerRef != nil && len(ownerRef) > 0 {
+		namespace, err := getNamespaceFromOwnerReference(pod.ObjectMeta.OwnerReferences[0])
+		if err != nil {
+			return pod, err
+		}
+		pod.ObjectMeta.Namespace = namespace
 	}
 	return pod, err
 }
 
+func getNamespaceFromOwnerReference(ownerRef metav1.OwnerReference) (namespace string, err error) {
+	namespace = ""
+	if ownerRef.Kind == "ReplicaSet" {
```

Comment:
I think switch case would be neater construct here instead of bunch of ' if else if'.

Reply:
done.